### PR TITLE
Mongodb connection setup refactoring

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do? 

Decouples environment config and other modules from mongodb setup. Now mongodb setup related code is segregated in `cc/setup/mongo` directory. 
Biggest functional change is that now pymongo connection is not established on import, but on explicit call during setup. 
Another functional change is that it's no longer a responsibility of `Environment` abstract class to contain and gather mongo parameters.

Add any further explanations here. 

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing? 
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by running UT's
    > Running the island with mongodb from island
    > Running the island with mongodb being started separatelly

### CODE REVIEW

File: monkey/monkey_island.py [hunk 1]
```diff
@@ -32,8 +31,7 @@
     # We need to initialize environment singleton before importing main,
     # because main imports modules from monkey_island/cc/models and models need a connection to the
     # mongodb. Mongodb connection parameters are initialized in environment singleton.
```
Comment (shreyamalviya) [inline monkey/monkey_island.py:33]: Stray comments

File: monkey/monkey_island.py [hunk 2]
```diff
@@ -32,8 +31,7 @@
...
-    env_singleton.initialize_from_file(server_config_path)
 
-    from monkey_island.cc.main import main  # noqa: E402
+    from monkey_island.cc.server_setup import setup_island  # noqa: E402
```
Comment (shreyamalviya) [inline monkey/monkey_island.py:35]: Do we still need to import this here? I think this can be moved to the top of the file now, right?

**Review by shreyamalviya — COMMENTED** [review]
What is `tests/unit_tests/monkey_island/cc/setup/test_process_runner.py` for?

File: monkey/monkey_island.py (hunk 1 above)
Comment (VakarisZ) [reply monkey/monkey_island.py:33]: Code folding 🤦‍♂️

**Comment by VakarisZ** [comment]
> What is `tests/unit_tests/monkey_island/cc/setup/test_process_runner.py` for?

It was a file that I removed. IIRC during merge I selected changes from local branch, but it didn't delete the file, just made it empty 🤷‍♂️ I'll keep an eye on these kinds of situations in the future.

**Review by shreyamalviya — APPROVED** [review]

**Pull request merged**